addon/tracked.ts

```diff
@@ -107,37 +105,31 @@ function installTrackedProperty(_target: object, key: string | symbol, descripto
 }
 
 
-function _tracked(target: object, key: string | symbol, descriptor?: PropertyDescriptor, dependencies?: string[]): PropertyDescriptor {
-  // descriptor is undefined for typescript class fields
-  if (descriptor === undefined || 'initializer' in descriptor) {
-    return installTrackedProperty(target, key, descriptor);
+function _tracked(
+  key: string | symbol,
+  descriptor: PropertyDescriptor,
+  initializer?: () => any,
+  dependencies?: string[]
+): PropertyDescriptor {
+  if (typeof descriptor.get !== 'function' && typeof descriptor.set !== 'function') {
+    return installTrackedProperty(key, descriptor, initializer);
   } else {
-    return descriptorForTrackedComputedProperty(target, key, descriptor, dependencies);
+    return descriptorForTrackedComputedProperty(key, descriptor, dependencies);
   }
 }
 
-// type CompatiblePropertyDecorator = (target: object, key: string | symbol, descriptor: PropertyDescriptor) => PropertyDescriptor;
-
-// @tracked
 // TODO: replace return w/ PropertyDescriptor once TS gets their decorator act together
-export function tracked(target: object, propertyKey: string | symbol, descriptor?: PropertyDescriptor): any;
-// @tracked('foo', 'bar')
-// TODO: replace return w/ CompatiblePropertyDecorator
-export function tracked(...args: string[]): any;
-// TODO: replace return w/ CompatiblePropertyDecorator | PropertyDescriptor
-export function tracked(
-  targetOrArgs: (string | object),
-  secondArg: string | symbol,
-  descriptorOrString: string | PropertyDescriptor | undefined,
-  ...rest: string[]): any
-  {
-  // if called for `@tracked('foo')`
-  if (typeof targetOrArgs === 'string') { //  @tracked('foo', 'bar')
-    const args =  [targetOrArgs, secondArg as string, descriptorOrString as string, ...rest];
-    return function(target: object, key: string | symbol, descriptor: PropertyDescriptor) {
-      return _tracked(target, key, descriptor, args);
-    }
-  } else { // @tracked
-    return _tracked(targetOrArgs, secondArg, descriptorOrString as PropertyDescriptor | undefined);
-  }
-}
+type TSDecorator = (target: object, propertyKey: string | symbol, descriptor?: PropertyDescriptor) => any;
+type TrackedDecorator = TSDecorator & ((...args: string[]) => TSDecorator);
+
+export const tracked: TrackedDecorator = decoratorWithParams((desc, params = []) => {
+  assert(`@tracked - Can only be used on class fields.`, desc.kind === 'field' || desc.kind === 'method');
```

Comment:
If you wanna be _really_ correct.

```suggestion
  assert(
    `@tracked - Can only be used on class fields.`,
    desc.kind === 'field' || desc.kind === 'method' && (
      typeof desc.descriptor.get === 'function' ||
      typeof desc.descriptor.det === 'function'
    )
  );
```